### PR TITLE
Refactor E2E templates generation

### LINKED ISSUE #804: Refactor templates for E2E
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
The cluster templates used by E2E can be built based on the flavours. This task refactors the existing structure using `Kustomize`.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
Refactors template generation for E2E tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #804 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Refactors template generation for e2e tests
```

### CODE REVIEW

**Comment by Amulyam24** [comment]
Changes applied, PTAL.

File: test/e2e/data/templates/cluster-template-md-remediation.yaml [hunk 1]
```diff
@@ -1,261 +1,259 @@
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
```
Comment (mkumatag) [inline test/e2e/data/templates/cluster-template-md-remediation.yaml:1]: do we still need this file?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: test/e2e/data/templates/cluster-template-md-remediation.yaml (hunk 1 above)
Comment (Amulyam24) [reply test/e2e/data/templates/cluster-template-md-remediation.yaml:1]: yes, this is what will finally be deployed in an E2E test - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/test/e2e/config/ibmcloud-e2e-powervs.yaml#L43

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**